client/me/get-apps/mobile-download-card.jsx

```diff
@@ -6,46 +6,263 @@
+			return element.numeric_code;
+		}
+
+		return null;
+	}
+
+	render() {
+		const translate = this.props.translate;
+
+		const has_all_data = this.props.hasUserSettings && this.props.hasLoadedAccountRecoveryPhone;
+		const { countryCode, number, isValid } = this.getPreferredNumber( has_all_data );
+		const { isMobile } = userAgent;
+		const feature_is_enabled = config.isEnabled( 'get-apps-sms' ) && ! isMobile;
+
+		return (
+			<Card className="get-apps__mobile">
+				<div className="get-apps__store-subpanel">
+					<div className="get-apps__card-text">
+						<h3 className="get-apps__card-title">{ translate( 'Mobile Apps' ) }</h3>
+						<p className="get-apps__description">
+							{ translate( 'WordPress at your fingertips.' ) }
+						</p>
+					</div>
+					<div className="get-apps__badges">
+						<AppsBadge
+							storeLink="https://play.google.com/store/apps/details?id=org.wordpress.android&referrer=utm_source%3Dcalypso-get-apps%26utm_medium%3Dweb%26utm_campaign%3Dmobile-download-promo-pages"
+							storeName={ 'android' }
+							titleText={ translate( 'Download the WordPress Android mobile app.' ) }
+							altText={ translate( 'Google Play Store download badge' ) }
+						/>
+						<AppsBadge
+							storeLink="https://itunes.apple.com/app/apple-store/id335703880?pt=299112&ct=calpyso-get-apps-button&mt=8"
+							storeName={ 'ios' }
+							titleText={ translate( 'Download the WordPress iOS mobile app.' ) }
+							altText={ translate( 'Apple App Store download badge' ) }
+						/>
+					</div>
+				</div>
+
+				{ feature_is_enabled ? (
+					<div className="get-apps__sms-subpanel">
+						<p>
+							<strong>{ translate( 'Ready to WordPress on the go?' ) }</strong>
+							<br />
+							{ translate(
+								'We’ll send you an SMS message with a download link for the right app for your mobile device.'
+							) }
+						</p>
+
+						<div className="get-apps__sms-field-wrapper">
+							{ has_all_data ? (
+								<FormPhoneInput
+									countriesList={ this.props.countriesList }
+									initialCountryCode={ countryCode }
+									initialPhoneNumber={ number }
+									phoneInputProps={ {
+										onKeyUp: this.onKeyUp,
+									} }
+									onChange={ this.onChange }
+								/>
+							) : (
+								<>
+									<QuerySmsCountries />
+									<QueryAccountRecoverySettings />
+									<QueryUserSettings />
+
+									<FormPhoneInput countriesList={ this.props.countriesList } isDisabled={ true } />
+								</>
+							) }
+						</div>
+						<div className="get-apps__sms-button-wrapper">
+							<p>{ translate( 'Standard SMS rates may apply' ) }</p>
+
+							<Button
+								className="get-apps__sms-button"
+								onClick={ this.onSubmit }
+								disabled={ ! isValid }
+							>
+								{ translate( 'Text me a link' ) }
+							</Button>
+						</div>
+					</div>
+				) : (
+					''
+				) }
+			</Card>
+		);
+	}
+
+	onChange = phoneNumber => {
+		this.setState( {
+			phoneNumber,
```

Comment:
I'd be careful to just copy the phoneNumber object given by `FormPhoneInput`'s `onChange` function. It seems the format is:
```
{
    phoneNumber,
    phoneNumberFull,
	countryData: { code, numeric_code, name, country_name }
    isValid
    validation: { error }
}
```
Which is different from the one you're returning in `getPreferredNumber`.

Reply:
I _think_ it still worked (at least it did in my manual test cases?), but you're totally right – to be safe, I've updated the `noPreferredNumber` and `state.phoneNumber` objects to use consistent keys with the values handpicked from the object provided by `FormPhoneInput`.